identity/managers/identifier.go

```diff
@@ -418,7 +418,7 @@ func (im *IdentifierIdentityManager) EndSession(ctx context.Context, rw http.Res
 	if esr.IDTokenHint != nil {
 		// Extended request, verify IDTokenHint and its claims if available.
 		esrClaims = esr.IDTokenHint.Claims.(*konnectoidc.IDTokenClaims)
-		clientDetails, err = im.clients.Lookup(ctx, esrClaims.Audience, "", esr.PostLogoutRedirectURI, origin, true)
+		clientDetails, err = im.clients.Lookup(ctx, esrClaims.Audience[0], "", esr.PostLogoutRedirectURI, origin, true)
```

Comment:
Is it guaranteed that `Audience` always has at least one element? I could not find anything which would suggest that, so I guess a check is needed (here and for the other places further down below in this PR).

Reply:
Ok. I'll try to address that.